### PR TITLE
bpo-46584: remove check for `py2.3` from `ctypes/test_python_api`

### PR DESCRIPTION
**Pull request by sobolevn**
https://bugs.python.org/issue46584

Automerge-Triggered-By: GH:zware

### CODE REVIEW

File: Lib/ctypes/test/test_python_api.py
Comment (arhadthedev) [inline Lib/ctypes/test/test_python_api.py]: Can `c_py_ssize_t` be removed altogether after it stopped to be a compatibility shim?

Comment (sobolevn) [reply Lib/ctypes/test/test_python_api.py]: Good idea! 👍

**Review by zware — APPROVED** [review]

**Comment by arhadthedev** [comment]
@mariatta Looks like miss-islington needs restoration of push rights for backport branches.

**Comment by AlexWaygood** [comment]
> @Mariatta Looks like miss-islington needs restoration of push rights for backport branches.

That's not the issue (miss-islington is trying to push to the main branch due to the "automerge" label being added). The issue is that Python is currently broken on the main branch, so all PRs are blocked until that's fixed (see https://mail.python.org/archives/list/python-dev@python.org/thread/HU2LH5MX44QZXZ36YS4SOG4ZUPT4E7QT/)

**Comment by zware** [comment]
Yep, we just need to wait.  I was just seeing whether @miss-islington would hold off until the branch was unblocked, or fail; it turned out to be fail (and holding off was probably an unrealistic expectation in the first place :)).  We'll get this merged when we can.

**Pull request merged**
